packages/libp2p/src/connection-manager/index.ts

```diff
@@ -584,8 +610,8 @@ export class DefaultConnectionManager implements ConnectionManager, Startable {
     }
 
     // check allow list
-    const allowConnection = this.allow.some(ma => {
-      return maConn.remoteAddr.toString().startsWith(ma.toString())
+    const allowConnection = this.allow.some(ipNet => {
+      return ipNet.contains(maConn.remoteAddr.nodeAddress().address)
```

Comment:
Prefer `toOptions` because `nodeAddress` will throw if the protocol isn't `tcp` or `udp`:

```suggestion
      return ipNet.contains(maConn.remoteAddr.toOptions().host)
```